kidkode/src/main/webapp/script.js

```diff
@@ -1,48 +1,151 @@
+    );
+  });
+
+  // finally combine our output list into one string of HTML and put it on the page
+  quizContainer.innerHTML = output.join("");
+}
+
+
+/*
+ * SHOW RESULTS
+ * First, we check whether the user picked "yes" or "no" as their choice for each
+ * question. Based on their choice, we add the careers under that choice into a 
+ * hashmap, and increment the points for the corresponding careers in the hashmap
+ */
+function showResults() {
+  // get answer containers from our quiz
+  const answerContainers = quizContainer.querySelectorAll(".answers");
+
+  // create empty hashmap for the user's results
+  let resultTracker = new Map();
+
+  // for each question interate through each carrer in points array...
+  myQuestions.forEach((currentQuestion, questionNumber) => {
+    const answerContainer = answerContainers[questionNumber];
+    const selector = `input[name=question${questionNumber}]:checked`;
+    const userAnswer = (answerContainer.querySelector(selector) || {}).value;
+
+    for (var i in currentQuestion.answers) {
+      if (userAnswer === currentQuestion.answers[i].choice) {
+        // ...and increment points for appropriate careers in hashmap
+        let pointsArr = currentQuestion.answers[i].points;
+        for (var j in pointsArr) {
+          let key = pointsArr[j];
+          resultTracker.set(key, getOrDefault(key, resultTracker) + 1);
+        }
+      }
     }
+  });
+
+  // create a variable to store the HTML output
+  const output = [];
+
+  // add results from hashmap into output
+  function createHTMLElements(value, key, resultTracker) {
+    output.push(
+      `<h2> ${key} </h2>
+        <h3> ${value} </h3>`
+    );
+  }
+  resultTracker.forEach(createHTMLElements);
+
+  // show total points for each category in hashmap
+  resultsContainer.innerHTML = output.join("");
 }
-/*example here for how updating the points in a hashmap would work.
-/ We would iterate through the points array in the current question's 
-object and update/add each career and point value to the hashmap*/
-$( document ).ready(function() {
-   let resultTracker = new Map();
-   let key = "gameArt";
-   result.set(key, getOrDefault(key, resultTracker) + 1);
-   console.log(getOrDefault(key, resultTracker));
-});
-
-  
\ No newline at end of file
+
+
+/*
+ * VARIABLES
+ */
+const quizContainer = document.getElementById("quiz");
+const resultsContainer = document.getElementById("results");
+const submitButton = document.getElementById("submit");
+const myQuestions = [
+  {
+    number: 0,
+    question: "Do you like to play videogames?",
+    answers: {
+      yes: {
+        choice: "yes",
+        points: ["gameDev", "animation", "2-dAnimation"],
+        next: 1
+      },
+      no: {
+        choice: "no",
+        points: ["intDesign", "webDesign", "UX"],
+        next: 2
+      }
+    },
+    previous: null
+  },
+  {
+    number: 1,
+    question: "Do you like to create art?",
+    answers: {
+      yes: {
+        choice: "yes",
```

Comment:
I like how you added this field to the example I suggested, I didn't think about how the choice field would make it easier!